overlord/snapstate/snapmgr.go

```diff
@@ -884,8 +883,13 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	if cand.Revision.Local() {
+		snapst.LocalRevision = ss.Revision()
+	}
+
```

Comment:
this looks wrong, we cannot backout out LocalRevision, another independent install may have already picked the next